### PR TITLE
Fix several bugs in interactive mode

### PR DESCRIPTION
**Pull request by raoyitao**
* Make it support Multitest-parts: 1> Should correctly set Multitest UID
  in returned JSON report; 2> Multitest-part name should be correctly
  displayed (with part information) on NavEntry.
* Deal with special symbol like "slash" in Multitest name or suite
  name, need to encode the URI strings and decode them on server side.
* For error result and unstable result, the NavEntry's color is still
  black, need to add some CSS style definition for interactive mode.
* If a test suite has 'setup' or 'teardown', after executing them the
  `runtime_status` should be set to "finished", or the test suite's
  `runtime_status` is still "running" and UI shows an incorrect icon.
* Update version of "immer" library.

## Checklist:
- [x] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [x] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: testplan/testing/multitest/base.py [hunk 1]
```diff
@@ -1033,7 +1032,7 @@ def _run_testcases_iter(self, testsuite, testcases):
         """
         pre_testcase = getattr(testsuite, "pre_testcase", None)
         post_testcase = getattr(testsuite, "post_testcase", None)
-        testsuite_uid = testsuite.name
+        testsuite_uid = testsuite.name  # Unique suite name in Multitest as UID
```
Comment (Pyifan) [inline testplan/testing/multitest/base.py:1035]: I think it is confusing to assign testsuite.name to another local variable called testsuite_uid. Can we simply use testsuite.name where it is needed?

File: testplan/web_ui/testing/src/Nav/InteractiveNavList.js [hunk 2]
```diff
@@ -14,7 +15,7 @@ const InteractiveNavList = (props) => {
     props,
     (entry) => (
       <InteractiveNavEntry
-        name={entry.name}
+        name={_.isEmpty(entry.part) ? entry.name : entry.uid}
```
Comment (Pyifan) [inline testplan/web_ui/testing/src/Nav/InteractiveNavList.js:18]: I thought this is before reset_uid() so we could always use entry.uid, no?

**Review by Pyifan — CHANGES_REQUESTED** [review]

File: testplan/runnable/interactive/http.py [hunk 3]
```diff
@@ -32,6 +34,17 @@
 from .reloader import ModuleReloader
 
 
+def decode_uri_component(func):
+    """Decode URI component before arguments passed to url route handler."""
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
```
[the remark this answers is not in the document]
Comment (raoyitao) [reply testplan/runnable/interactive/http.py:41]: I think this is guaranteed by @api.route that when a class is decorated, its 'get' and 'set' methods are processed, they can receive an instance as args[0], and several named arguments as kwargs.

File: testplan/web_ui/testing/src/Nav/InteractiveNavList.js (hunk 2 above)
Comment (raoyitao) [reply testplan/web_ui/testing/src/Nav/InteractiveNavList.js:18]: seems we can use entry.uid

File: testplan/testing/multitest/base.py (hunk 1 above)
Comment (raoyitao) [reply testplan/testing/multitest/base.py:1035]: I think the author want to emphasize that unique id is used as identifier when transferred through network. let me think about it again.

**Review by Pyifan — DISMISSED** [review]

**Review by Pyifan — APPROVED** [review]

**Pull request merged**